collection-scripts/gather_network_logs

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# start off by doing a gather, which includes gather_network_logs_basics.
+/usr/bin/gather
+
+BASE_COLLECTION_PATH="must-gather"
+NETWORK_LOG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/network_logs"}
+OVNK_EXTRAS_STORE_PATH=${OUT:-"ovnk_extras_store"}
+
+mkdir -p "${NETWORK_LOG_PATH}"/
+
+function gather_extra_ovn_kubernetes_data_interconnect_mode {
+  echo "INFO: Gathering ovn-kubernetes extra logs"
+  OVNKUBE_CONTROLLER_PODS=($(oc -n openshift-ovn-kubernetes get pods -l app=ovnkube-node -o=jsonpath='{.items[*].metadata.name}'))
+  # collect additional ovnkube logs from each node
+  CONTAINERS="ovnkube-controller"
+  for CONTAINER in "${CONTAINERS}" ; do
+    for OVNKUBE_CONTROLLER_POD in "${OVNKUBE_CONTROLLER_PODS[@]}"; do
+      echo "INFO: Gathering "${CONTAINER}" extra logs from "${OVNKUBE_CONTROLLER_POD}"..."
+      DEST_DIR="${NETWORK_LOG_PATH}/${OVNK_EXTRAS_STORE_PATH}/${OVNKUBE_CONTROLLER_POD}/${CONTAINER}"
+      mkdir -p "${DEST_DIR}"
+      oc rsync --include='*log*' --exclude='*' -n openshift-ovn-kubernetes -c ${CONTAINER} \
+        ${OVNKUBE_CONTROLLER_POD}:/var/log/ovnkube/ ${DEST_DIR} 2>&1 & PIDS+=($!)
+    done
+  done
+}
+
+function gather_extra_ovn_kubernetes_data {
+  sample_node=$(oc get no -o jsonpath='{.items[0].metadata.name}')
+  sample_node_zone=$(oc get node "${sample_node}" -o jsonpath='{.metadata.annotations.k8s\.ovn\.org/zone-name}')
+  if [ "${sample_node}" = "${sample_node_zone}" ]; then
```

Comment:
in which case we expect this condition to be false?

Reply:
backwards compatibility makes perfect sense! I am just trying to figure out if we only need to collect extra logs in 1-node-per-zone config (e.g. is there a possible default zone or some other type of config where we would have libovsdb logs in another file)?

Reply:
ah. ic. no, iirc there is no space issues when we only have 3 ovnkube-controllers, so nothing additional here for non-ovn-ic cases.